types/domexception/index.d.ts

```diff
@@ -1,7 +1,71 @@
-// Type definitions for domexception 2.0
+// Type definitions for domexception 4.0
 // Project: https://github.com/jsdom/domexception#readme
 // Definitions by: ExE Boss <https://github.com/ExE-Boss>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.0
 
 export = DOMException;
+
+/**
+ * An abnormal event (called an exception) which occurs as a result of calling
+ * a method or accessing a property of a web API.
+ */
+declare class DOMException extends Error {
+    readonly code: number;
+    readonly message: string;
+    readonly name: string;
+
+    constructor(message?: string, name?: string);
+
+    readonly INDEX_SIZE_ERR: 1;
```

Comment:
Perhaps instead we should copy TypeScript's type?

https://github.com/microsoft/TypeScript/blob/92af654a83c497eb35aed7d186b746c8ca4b88fb/src/lib/dom.generated.d.ts#L3681

Reply:
I updated it to exactly match TypeScript's lib.dom type so that the two are interchangeable.

Reply:
They will still be compatible even if this one uses a `class` (but with the constants being `number`)